pkg/client/kubernetes/apply.go

```diff
@@ -118,62 +117,58 @@ func (c *Applier) deleteObject(ctx context.Context, desired *unstructured.Unstru
 }
 
 // DefaultApplierOptions contains options for common k8s objects, e.g. Service, ServiceAccount.
-var DefaultApplierOptions = ApplierOptions{
-	MergeFuncs: map[schema.GroupKind]MergeFunc{
-		corev1.SchemeGroupVersion.WithKind("Service").GroupKind(): func(newObj, oldObj *unstructured.Unstructured) {
-			// We do not want to overwrite a Service's `.spec.clusterIP', `.spec.healthCheckNodePort` or '.spec.ports[*].nodePort' values.
-			oldPorts := oldObj.Object["spec"].(map[string]interface{})["ports"].([]interface{})
-			newPorts := newObj.Object["spec"].(map[string]interface{})["ports"].([]interface{})
-			ports := []map[string]interface{}{}
-
-			// Check whether ports of the newObj have also been present previously. If yes, take the nodePort
-			// of the existing object.
-			for _, newPort := range newPorts {
-				np := newPort.(map[string]interface{})
-
-				for _, oldPort := range oldPorts {
-					op := oldPort.(map[string]interface{})
-					// np["port"] is of type float64 (due to Helm Tiller rendering) while op["port"] is of type int64.
-					// Equality can only be checked via their string representations.
-					if fmt.Sprintf("%v", np["port"]) == fmt.Sprintf("%v", op["port"]) {
-						if nodePort, ok := op["nodePort"]; ok {
-							np["nodePort"] = nodePort
-						}
+var DefaultApplierOptions = MergeFuncs{
```

Comment:
Do you think it make sense to keep the name `DefaultApplierOptions` although it is not only restricted to MergeFunc? Maybe `DefaultMergeFuncs` would make more sense?